source/source.go

```diff
@@ -33,3 +43,47 @@ const (
 type Source interface {
 	Endpoints() ([]*endpoint.Endpoint, error)
 }
+
+// Config holds shared configuration options for all Sources.
+type Config struct {
+	KubeClient    kubernetes.Interface
```

Comment:
I personally think `KubeClient` does not belong to config and `Config` should only be used in `store.go` for initialisation, which would also mean renaming `Config` -> `StoreConfig`